client/src/__tests__/container.test.js

```diff
@@ -12,9 +13,105 @@ it('should render correctly with no props', () => {
   expect(container).toMatchSnapshot();
 });
 
-it('should call getReviews during componentDidMount', () => {
-  const getReviews = jest.spyOn(Container.prototype, 'getReviews');
+it('should call getData during componentDidMount', () => {
+  const getData = jest.spyOn(Container.prototype, 'getData');
   const container = mount(<Container />);
 
-  expect(getReviews).toHaveBeenCalledTimes(1);
-});
\ No newline at end of file
+  expect(getData).toHaveBeenCalledTimes(1);
+});
+
+// it('should display the modal when showModal state is true', () =>{
+//   const container = mount(<Container />);
+//   container.setState({modalReview: 0, showModal: true});
+
+//   console.log(container.find('.modal').debug());
+
+//   expect(container.find('.modal').exists()).toEqual(true);
+// });
+
+// it('should close the modal on button click', () => {
+//   const closeModal = jest.spyOn(Container.prototype, 'closeModal');
+//   const container = mount(<Container />);
+//   container.setState({showModal: true});
+//   container.find('.closeModal').simulate('click');
+
+//   expect(closeModal).toHaveBeenCalledTimes(1);
+// });
+
+// it('should display average stars based on function call', () => {
+//   const getAverageStars = jest.spyOn(Container.prototype, 'getAverageStars');
+//   const container = shallow(<Container />);
+//   container.instance().getAverageStars(reviews);
+//   container.update();
+
+//   console.log(container.find('.averageStars').debug());
+
+//   expect(container.find('.averageStars')).toHaveLength(3);
+// });
+
+it('should set the ReviewContainer limit to 20 when showMore is clicked', () => {
+  const showMore = jest.spyOn(Container.prototype, 'showMore');
+  const container = shallow(<Container />);
+  container.instance().showMore();
+  container.update();
```

Comment:
Likewise, I didn't know you could do these commands here.

Reply:
`.instance().methodName()` is how you invoke the method in the test. `.update()` will update the component if state changed